### ISSUE TITLE
reset form setting values nested in an array to undefined (v4.0.5)

### ISSUE THREAD

**Issue by rhythnic** [erikras/redux-form#475]
I have a form which contains the fields:
```
'group.members[].name',
'group.members[].age'
```
I am setting initialValues, which is a deep JS object, through the mapStateToProps function
```
// initialValues
{
  group: {
    members: [{name: 'Bob', age: 40}]
  }
}
```
The form is properly initialized;
```
// this.props.fields
{group:
    members: [
      {
        name:  {defaultValue: 'Bob', initialValue: 'Bob', value: 'Bob'},
        age:  {defaultValue: 40, initialValue: 40, value: 40}
      }
      ...
   ]
```
however, after resetting the form, the values are set to undefined, instead of being set to the initialValue.
```
// this.props.fields
{group:
    members: [
      {
        name:  {defaultValue: 'Bob', initialValue: 'Bob', value: undefined},
        age:  {defaultValue: 40, initialValue: 40, value: undefined}
      }
      ...
   ]
```

**Comment by rhythnic** [comment]
I'm working around the problem by destroying the form and reinitializing.
I'm using the 'data' property instead of 'initialValues' because the initialValues prop seems to be intercepted by Redux Form and isn't available inside of componentWillReceiveProps.
```
componentWillReceiveProps(nextProps) {
  if (this.props.editing && !nextProps.editing) {
    this.props.destroyForm();
    this.props.initializeForm(this.props.data);
  }
}
```

**Comment by rhythnic** [comment]
It occurred to me that the name of the properties, which I changed in the example, might be the reason.  The properties are actually 'name' and 'value'.  I realized it when using getValues() returned the value of the 'value' property as the value of the whole object.  I'm going to close this issue and open up a [new issue that focuses on properties named 'value'](https://github.com/erikras/redux-form/issues/477).
